mitosheet/src/components/LoadingIndicator.tsx

```diff
@@ -1,43 +1,275 @@
-        }, 500);
-    }, []);
-
-    // We start the indicator at -1, so that we don't display anything
-    // for the first half second. This makes us only display the indicator
-    // for actually long running operations.
-    if (!display) {
+        const interval = setInterval(() => {
+            const messagesToDisplay = getMessageTypesToDisplay(props.loading);
+            if (messagesToDisplay.length === 0) {
+                setCurrentLoadingMessage(undefined);
+            } else {
+                setCurrentLoadingMessage(prevLoadingMessage => {
+                    const topMessageID = messagesToDisplay[0][1];
+                    if (prevLoadingMessage === undefined || topMessageID !== prevLoadingMessage[1]) {
+                        return [Date.now(), topMessageID];
+                    }
+                    return prevLoadingMessage;
+                })
+            }
+            // We always refresh this state, though, so that this rerenders 
+            // so that we can refresh the messaging to the user
+            setCurrentLoadingMessage(prevCurrentLoadingMessage => {
+                if (prevCurrentLoadingMessage === undefined) return prevCurrentLoadingMessage;
+                return [prevCurrentLoadingMessage[0], prevCurrentLoadingMessage[1]];
+            })
+        }, 1000)
+        return () => {clearInterval(interval)};
+    }, [props.loading])
+
+
+    const messagesToDisplay = getMessageTypesToDisplay(props.loading);
+
+    if (!display || messagesToDisplay.length === 0) {
         return <React.Fragment/>
     }
 
     return (
         <div className='loading-indicator-container'>
-            <p className='ml-20px'>
-                Loading<LoadingDots/>
+            <p className='loading-indicator-header text-header-3 text-color-white-important'>
+                Processing {messagesToDisplay.length} edit{messagesToDisplay.length <= 1 ? '' : 's'}
             </p>
+            <div className='loading-indicator-content'>
+                {messagesToDisplay.map((([messageType, message_id], index) => {
+                    const is10SecondsAgo = currentLoadingMessage && message_id === currentLoadingMessage[1] && (currentLoadingMessage[0] <= Date.now() - 10 * 1000);
+
+                    return (messageType !== undefined && 
+                        <div 
+                            key={index} 
+                            className={classNames('loading-indicator-item', 'text-body-1', 'mb-5px', 'mt-5px', {'text-color-medium-gray-important': index !== 0})}
+                        >
+                            <div className='loading-indicator-icon' style={{opacity: index !== 0 ? '50%' : undefined}}>
+                                {getIcon(messageType, '15', '15')}
+                            </div>
+                            <div className='ml-20px'>
+                                {getDisplayMessageForMessageType(messageType)} {is10SecondsAgo ? '... still working' : ''}
```

Comment:
Instead of having this appear in the same line as the message, can we have it underneath the message original message in the subtext font so that it stands out a bit more. 

And then since its no longer attatched inline, maybe we can change it to "Still executing pandas code". 

<img width="441" alt="Screen Shot 2022-06-20 at 11 09 46 PM" src="https://user-images.githubusercontent.com/18709905/174708307-38643b55-75b1-49e8-8abb-41649529664a.png">